mlflow/store/dbmodels/models.py

```diff
@@ -45,11 +45,12 @@ def _create_entity(base, model):
                     metrics = {}
                     for o in obj:
                         existing_metric = metrics.get(o.key)
-                        if (existing_metric is None) or (o.timestamp > existing_metric.timestamp)\
-                            or (o.timestamp == existing_metric.timestamp
-                                and o.value > existing_metric.value):
-                            metrics[o.key] = Metric(o.key, o.value, o.timestamp)
-                    obj = metrics.values()
+                        if (existing_metric is None)\
+                            or ((o.step, o.timestamp, o.value) >=
+                                (existing_metric.step, existing_metric.timestamp,
+                                 existing_metric.value)):
+                            metrics[o.key] = Metric(o.key, o.value, o.timestamp, o.step)
+                    obj = list(metrics.values())
```

Comment:
Was the set return type wrong before?

Reply:
Good question, in Python 2 dict.values() is a list of `Metric` objects but in Python 3 dict.values() is an object of type `dict_values`, which supports iteration but not lookup by index. From the perspective of RunData there's no difference (because RunData only ever iterates over the `metrics` constructor argument), but having `dict_values` instead of a list broke one of my unit tests where I try to do an index lookup ([link](https://github.com/mlflow/mlflow/pull/1142/files/0d9e49878b1c422f363cbb042d5ba4c30818eb2b#diff-477cd83ba3c2576bc1540a7d61d60045R318))